cpp/cmake_modules/ThirdpartyToolchain.cmake

```diff
@@ -4485,115 +4485,120 @@ target_include_directories(arrow::hadoop INTERFACE "${HADOOP_HOME}/include")
+  get_target_property(LZ4_INCLUDE_DIR LZ4::lz4 INTERFACE_INCLUDE_DIRECTORIES)
+  get_filename_component(LZ4_ROOT "${LZ4_INCLUDE_DIR}" DIRECTORY)
+  set(LZ4_HOME
+      ${LZ4_ROOT}
+      CACHE BOOL "" FORCE)
+  set(LZ4_LIBRARY $<TARGET_FILE:LZ4::lz4>)
+  set(LZ4_STATIC_LIB $<TARGET_FILE:LZ4::lz4>)
 
-  get_target_property(ORC_PROTOBUF_ROOT ${ARROW_PROTOBUF_LIBPROTOBUF}
+  get_target_property(PROTOBUF_INCLUDE_DIR ${ARROW_PROTOBUF_LIBPROTOBUF}
                       INTERFACE_INCLUDE_DIRECTORIES)
-  get_filename_component(ORC_PROTOBUF_ROOT "${ORC_PROTOBUF_ROOT}" DIRECTORY)
-
-  get_target_property(ORC_SNAPPY_INCLUDE_DIR ${Snappy_TARGET}
+  get_filename_component(Protobuf_ROOT "${PROTOBUF_INCLUDE_DIR}" DIRECTORY)
+  set(PROTOBUF_HOME
+      ${Protobuf_ROOT}
+      CACHE BOOL "" FORCE)
+  target_include_directories(${ARROW_PROTOBUF_LIBPROTOC}
+                             INTERFACE "${PROTOBUF_INCLUDE_DIR}")
+  set(PROTOBUF_EXECUTABLE $<TARGET_FILE:${ARROW_PROTOBUF_PROTOC}>)
+  set(PROTOBUF_LIBRARY $<TARGET_FILE:${ARROW_PROTOBUF_LIBPROTOBUF}>)
+  set(PROTOC_LIBRARY $<TARGET_FILE:${ARROW_PROTOBUF_LIBPROTOC}>)
+
+  get_target_property(SNAPPY_INCLUDE_DIR ${Snappy_TARGET} INTERFACE_INCLUDE_DIRECTORIES)
+  get_filename_component(Snappy_ROOT "${SNAPPY_INCLUDE_DIR}" DIRECTORY)
+  set(SNAPPY_HOME
+      ${Snappy_ROOT}
+      CACHE BOOL "" FORCE)
+  set(SNAPPY_LIBRARY
+      $<TARGET_FILE:${Snappy_TARGET}>
+      CACHE STRING "" FORCE)
+
+  get_target_property(ZLIB_INCLUDE_DIR ZLIB::ZLIB INTERFACE_INCLUDE_DIRECTORIES)
+  get_filename_component(ZLIB_ROOT "${ZLIB_INCLUDE_DIR}" DIRECTORY)
+  set(ZLIB_HOME
+      ${ZLIB_ROOT}
+      CACHE BOOL "" FORCE)
+
+  get_target_property(ZSTD_INCLUDE_DIR ${ARROW_ZSTD_LIBZSTD}
                       INTERFACE_INCLUDE_DIRECTORIES)
-  get_filename_component(ORC_SNAPPY_ROOT "${ORC_SNAPPY_INCLUDE_DIR}" DIRECTORY)
-
-  get_target_property(ORC_LZ4_ROOT LZ4::lz4 INTERFACE_INCLUDE_DIRECTORIES)
-  get_filename_component(ORC_LZ4_ROOT "${ORC_LZ4_ROOT}" DIRECTORY)
+  get_filename_component(ZSTD_ROOT "${ZSTD_INCLUDE_DIR}" DIRECTORY)
+  set(ZSTD_HOME
+      ${ZSTD_ROOT}
+      CACHE BOOL "" FORCE)
+  set(ZSTD_LIBRARY $<TARGET_FILE:${ARROW_ZSTD_LIBZSTD}>)
+  set(ZSTD_STATIC_LIBRARY $<TARGET_FILE:${ARROW_ZSTD_LIBZSTD}>)
+
+  set(BUILD_CPP_TESTS
+      OFF
+      CACHE BOOL "" FORCE)
+  set(BUILD_JAVA
+      OFF
+      CACHE BOOL "" FORCE)
+  set(BUILD_LIBHDFSPP
+      OFF
+      CACHE BOOL "" FORCE)
+  set(BUILD_TOOLS
+      OFF
+      CACHE BOOL "" FORCE)
+  set(INSTALL_VENDORED_LIBS
+      OFF
+      CACHE BOOL "" FORCE)
+  set(STOP_BUILD_ON_WARNING
+      OFF
+      CACHE BOOL "" FORCE)
 
-  get_target_property(ORC_ZSTD_ROOT ${ARROW_ZSTD_LIBZSTD} INTERFACE_INCLUDE_DIRECTORIES)
-  get_filename_component(ORC_ZSTD_ROOT "${ORC_ZSTD_ROOT}" DIRECTORY)
+  # TODO: This should be fixed in upstream.
+  list(PREPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_BINARY_DIR}/_deps/orc-src/cmake_modules)
```

Comment:
@wgtmac If we don't have this, ORC uses the Arrow's `cpp/cmake_modules/ThirdpartyToolchain.cmake` not ORC's `cmake_modules/ThirdpartyToolchian.cmake`.

Here is a fix of this in ORC:  

```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 213f55721..baf5640b4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,7 +30,7 @@ SET(CPACK_PACKAGE_VERSION_MAJOR "2")
 SET(CPACK_PACKAGE_VERSION_MINOR "1")
 SET(CPACK_PACKAGE_VERSION_PATCH "0-SNAPSHOT")
 SET(ORC_VERSION "${CPACK_PACKAGE_VERSION_MAJOR}.${CPACK_PACKAGE_VERSION_MINOR}.${CPACK_PACKAGE_VERSION_PATCH}")
-set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake_modules")
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_CURRENT_SOURCE_DIR}/cmake_modules")
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON) # For clang-tidy.
 
 option (BUILD_JAVA
```

Reply:
Thanks! I've created https://github.com/apache/orc/pull/1959